lms/djangoapps/instructor_analytics/basic.py

```diff
@@ -351,11 +353,20 @@ def extract_details(exam_attempt, features):
                 u'{status} Comments'.format(status=status): '; '.join(comment_list),
             })
 
+        proctored_exam['track'] = course_enrollment.filter(user_id=exam_attempt['user_id'])[0].mode
```

Comment:
This still does executes a query for each row. Django querysets are lazily evaluated: your first construction `CourseEnrollment.objects.filter(course_id= CourseKey.from_string(exam_attempts[0]['course_id']))` doesn't execute a query. It starts to build a query. When you do a further `filter(...)`, it modifies the query. When you slice the queryset `[0]`, it has to execute the query to return the object. But that second filter makes it a different query every time.